main.py

```
@@ -423,26 +557,46 @@ async def cb_admin_check_db(callback: CallbackQuery):
 
-    @dp.message(AdminStates.delete_account_input)
+    @dp.message(AdminStates.delete_account_input, F.text)
     async def step_admin_delete_account(message: Message, state: FSMContext):
+        if not message.text:
+            await message.answer("❌ Пожалуйста, отправьте текстовое сообщение.")
+            return
         email = message.text.strip()
         
-        if not validate_email(email):
-            await message.answer("❌ Некорректный e-mail")
+        # Улучшенная валидация email с проверкой провайдера
```

Ruff (0.14.10)

Comment contains ambiguous `с` (CYRILLIC SMALL LETTER ES). Did you mean `c` (LATIN SMALL LETTER C)?

(RUF003)